apps/rig/lib/result.ex

```diff
@@ -0,0 +1,273 @@
+defmodule Result do
```

Comment:
Me too, but I think I have to try it on my own to understand completely 👍 

What about publishing this as a separate hex.package?

I did a quick check and there were already discussions around it:
https://elixirforum.com/t/is-there-a-better-way-to-handle-ok-tuples/14178/21

And discussing it referencing Rust:
https://elixirforum.com/t/usage-of-ok-result-error-vs-some-result-none/3253/9

Seems like there is already an existing package:
https://hex.pm/packages/exceptional
https://hex.pm/packages/ok

I'm not yet sure if those solve the same issue. Anyways if you think about it, you can consider those

Reply:
I've thought about publishing it as well, but I thought it would be good to stabilize it within RIG first. I know about the "ok" package, but I wanted something simpler. Also, I think that if we rely on this code throughout the code base, we should fully understand and maintain it ourselves.

Reply:
actually, there is already a package with the same and a similar goal: https://hex.pm/packages/result
I still like my implementation better :)